### ISSUE TITLE
Agent tracing warning for IndexOutOfBoundsException

### ISSUE THREAD

**Issue by sendev1** [oracle/graal#2826]
I am using native-image agent to generate the json configs but while running the native agent I see so many `IndexOutOfBoundsException` warning all over the logs for so many class. Sample is below. I suspect that this would have severe impact down the road, so reaching out to get some help on how to fix them

```
WARNING: Error processing trace entry: java.lang.IndexOutOfBoundsException: start 2, end 1, s.length() 2: {caller_class=org.springframework.boot.context.properties.bind.Bindable, result=true, args=[[Z], function=newInstance, tracer=reflect, class=java.lang.reflect.Array}
WARNING: Error processing trace entry: java.lang.IndexOutOfBoundsException: start 2, end 1, s.length() 2: {caller_class=org.springframework.boot.context.properties.bind.Bindable, result=true, args=[[I], function=newInstance, tracer=reflect, class=java.lang.reflect.Array}
WARNING: Error processing trace entry: java.lang.IndexOutOfBoundsException: start 2, end 1, s.length() 2: {caller_class=org.springframework.boot.context.properties.bind.Bindable, result=true, args=[[I], function=newInstance, tracer=reflect, class=java.lang.reflect.Array}

WARNING: Error processing trace entry: java.lang.IndexOutOfBoundsException: start 2, end 1, s.length() 2: {caller_class=org.apache.cxf.common.util.ProxyHelper, result=true, args=[[C], function=forName, tracer=reflect, class=java.lang.Class}
WARNING: Error processing trace entry: java.lang.IndexOutOfBoundsException: start 2, end 1, s.length() 2: {caller_class=org.apache.cxf.common.util.ProxyHelper, result=true, args=[[C], function=forName, tracer=reflect, class=java.lang.Class}
```

**Comment by mcraj017** [comment]
@sendev1 please provide the reproducer

**Comment by gradinac** [comment]
Hey @sendev1, I think https://github.com/oracle/graal/commit/551bcea220209a91cdd06a398679cad3cd77b7c7 contains a fix for the issues you are facing - could you verify if the error still occurs with the latest GraalVM dev release?

**Comment by cstancu** [comment]
Please reopen if the above-mentioned changeset doesn't fix the issue.

**Comment by andreaTP** [comment]
Please reopen, this is an issue with Graal 20.2.0 ( @cstancu @gradinac )
Here you have a repro:

https://github.com/andreaTP/graal-repro-2826

**Comment by gradinac** [comment]
Hey @andreaTP! Yes, this issue has been verified with GraalVM 20.2 and has been addressed for the next release (GraalVM 20.3). Since GraalVM 20.3 is not out yet, you could try if the latest dev build fixes it for you: https://github.com/graalvm/graalvm-ce-dev-builds/releases

**Comment by andreaTP** [comment]
Thanks for the heads up @gradinac ! I'll test as soon as 20.3 is out 😄
